### PR TITLE
feat(contrib): retry on HTTP 405 error from Arsenal

### PR DESCRIPTION
**Pull request by fsamin**

### CODE REVIEW

**Comment by ovh-cds** [comment]
CDS Report build-cds#21376.0 ✔ 
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✔ 
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#21376.0 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✘  
  * Standalone Tests v2 ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#21376.2 ✘  
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✘

File: contrib/integrations/arsenal/arsenal.go [hunk 1]
```diff
@@ -92,12 +92,25 @@ func (ac *Client) Deploy(deployRequest *DeployRequest) (*DeployResponse, error)
...
+		statusCode, rawBody, err := ac.doRequest(req, deployResult)
+		if err != nil {
+			return nil, err
+		}
+		if statusCode == http.StatusOK {
+			return deployResult, lastErr
+		}
+		if statusCode == http.StatusMethodNotAllowed {
+			lastErr = &RequestError{fmt.Sprintf("deploy request failed (HTTP status %d): %s", statusCode, rawBody)}
+			time.Sleep(100 * time.Second)
```
Comment (sguiheux) [inline contrib/integrations/arsenal/arsenal.go:111]: Why 100s ?

**Review by sguiheux — CHANGES_REQUESTED** [review]
Maybe update also the action deployArsenal

**Comment by ovh-cds** [comment]
CDS Report build-cds#21382.0 ✔ 
* 
  * Build and Package All -   
  * Build and Package Contrib Action ✔ 
  * Build and Package Contrib Code Plugin ✔ 
  * Build and Package Contrib Integrations ✔ 
  * Build and Package Engine, Worker, CLI and Doc ✔ 
  * Build and Package UI ✔ 
  * Build Debpacker ✔ 
  * Build SMTPMock ✔ 
  * Lint UI Translate ✔ 
  * Test CLI ✔ 
  * Test Contrib ✔ 
  * Test Engine ✔ 
  * Test SDK ✔ 
  * Test UI ✔

**Comment by ovh-cds** [comment]
CDS Report integration-cds#21382.0 ✔ 
* Integration
  * Docker Compose Tests ✔ 
  * Standalone Tests ✔ 
  * Standalone Tests v1 ✔ 
  * Standalone Tests v2 ✔

**Review by sguiheux — APPROVED** [review]

**Pull request merged**